app.js

```diff
@@ -560,6 +561,25 @@ var eventsModule = function () {
             //   values (CSV), into a hierarchy using d3.stratify.
             var root = EFsToHierarchy();
 
+            var efDurationSum = function(efNode) {
+                if (efNode.children === undefined) {
+                    // Node has no children, return duration value
+                    return efNode.data.durationMinutes;
+                } else {
+                    // Node has children, compute duration of each child
+                    var childSum = 0;
+                    for (var i = 0; i < efNode.children.length; i++) {
+                        childSum += efDurationSum(efNode.children[i]);
+                    }
+                    return childSum;
+                }
+            }
+
+            var totalTime = efDurationSum(root);
```

Comment:
If you move that implementation into the `eachBefore` function, then i think you could call `durationMinutes` straight of the root's data property. Might be simpler.